doc/API/overview.rst

```diff
@@ -92,39 +92,49 @@ When PennyLane needs to evaluate a QNode, it accesses the :meth:`~.Device.execut
 
 .. code-block:: python
 
+    results = []
+
     with self.execution_context():
         self.pre_apply()
         for operation in queue:
             self.apply(operation.name, operation.wires, operation.parameters)
         self.post_apply()
 
-        self.pre_expval()
-        expectations = [self.expval(e.name, e.wires, e.parameters) for e in expectation]
-        self.post_expval()
+        self.pre_measure()
+
+        for obs in observables:
+            if obs.return_type == "expectation":
+                results.append(self.expval(obs.name, obs.wires, obs.parameters))
+            elif obs.return_type == "variance":
+                results.append(self.var(obs.name, obs.wires, obs.parameters))
 
-        return np.array(expectations)
+        self.post_measure()
 
-where ``queue`` is a list of PennyLane :class:`~.Operation` instances to be applied, and ``expectation`` is a list of PennyLane :class:`~.Expectation` instances to be measured and returned. In most cases, there are therefore a minimum of two methods that any device **must** implement:
+        return np.array(results)
+
+where ``queue`` is a list of PennyLane :class:`~.Operation` instances to be applied, and ``observables`` is a list of PennyLane :class:`~.Observable` instances to be measured and returned. In most cases, there are therefore a minimum of three methods that any device **must** implement:
 
 * :meth:`~.Device.apply`: This accepts an operation name (as a string), the wires (subsystems) to apply the operation to, and the parameters for the operation, and should apply the resulting operation to given wires of the device.
 
 * :meth:`~.Device.expval`: This accepts an observable name (as a string), the wires (subsystems) to measure, and the parameters for observable. It is expected to return the resulting expectation value from the device.
 
-  .. note:: Currently, PennyLane only supports expectations that return a scalar value.
+* :meth:`~.Device.var`: This accepts an observable name (as a string), the wires (subsystems) to measure, and the parameters for observable. It is expected to return the resulting variance of the measured observable value from the device.
```

Comment:
I changed it to not be an abstract method in the `Device` class. If the user calls `var()` and it is not implemented by the device, it will just raise a `NotImplementedError`.